packages/phone-number-privacy/common/src/utils/authentication.ts

```diff
@@ -63,6 +63,24 @@ export async function authenticateUser(
   return verifySignature(message, messageSignature, signer)
 }
 
+export function getMessageDigest(message: string) {
+  // NOTE: Elliptic will truncate the raw msg to 64 bytes before signing,
+  // so make sure to always pass the hex encoded msgDigest instead.
+  return crypto.createHash('sha256').update(JSON.stringify(message)).digest('hex')
+}
+
+export function signWithRawDEK(msg: string, rawKey: string) {
```

Comment:
Per my other comment, it seems like this is not a DEK specific function. Maybe we can call this signWithRawKey and just import and export it without changes from the identity pkg for backwards compatibility?

Reply:
It's not strictly specific to DEKs (neither is `verifyDEKSignature` in authentication.ts), but it does seem to be used exclusively with DEKs in the monorepo (otherwise, other signing tooling is generally preferred/used). That said, I do like just import/exporting it -- changing the name but adding a comment that this is the counterpart for `verifyDEKSignature`; lmk if that sounds ok to you!

Side note: do you have context on why we have custom sign/verify functions in ODIS at all as opposed to using `SignatureUtils`?

Reply:
Sounds good! I honestly do not haha. It's been that way for a while and I vaguely remember there was some reason but I'm not confident. I added a TODO to revisit though!